src/cocoa/toga_cocoa/widgets/internal/cells.py

```diff
@@ -24,31 +28,38 @@
     NSLayoutConstraint,
     NSLayoutRelationEqual,
     NSLineBreakMode,
+    NSButton,
+    NSSwitchButton,
     at,
     objc_method,
 )
 
 
-class TogaIconView(NSTableCellView):
+class TogaTableCellView(NSTableCellView):
+
+    checkbox = objc_ivar(objc_id)
 
     @objc_method
-    def setup(self):
-        image_view = NSImageView.alloc().init()
-        text_field = NSTextField.alloc().init()
+    def initWithLayout(self):
+        send_super(__class__, self, "init")
```

Comment:
Handling init this way is cleaner but comes with a significant performance impact. Though I prefer having a clean implementation here and working to rubicon-objc to improve the performance.